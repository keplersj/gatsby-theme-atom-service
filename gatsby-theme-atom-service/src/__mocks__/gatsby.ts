import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
    }): React.ReactElement =>
      React.createElement("a", {
        ...rest,
        href: to
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
