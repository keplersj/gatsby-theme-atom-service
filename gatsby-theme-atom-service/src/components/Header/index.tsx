import { graphql, useStaticQuery } from "gatsby";
import * as React from "react";
import { Container, makeStyles, Typography } from "@material-ui/core";

const useStyles = makeStyles({
  banner: {
    background: "#efeae1",
    textAlign: "center",
    padding: "2em"
  }
});

const Header = (): React.ReactElement => {
  const classes = useStyles({});
  const data = useStaticQuery(graphql`
    query HeroQuery {
      site {
        siteMetadata {
          title
          description
        }
      }
      logo: file(name: { eq: "logo" }) {
        publicURL
      }
    }
  `);

  return (
    <Container className={classes.banner} maxWidth="xl">
      <img src={data.logo.publicURL} width="225em" height="225em" alt="" />
      <Typography variant="h2">{data.site.siteMetadata.title}</Typography>
      <Typography variant="h4">{data.site.siteMetadata.description}</Typography>
    </Container>
  );
};

export default Header;
