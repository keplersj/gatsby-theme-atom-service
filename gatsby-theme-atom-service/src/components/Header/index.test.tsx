import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Header from ".";

describe("Header", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: "Example Site",
          description: "This is an example site description.",
        },
      },
      logo: {
        publicURL: "/static/logo.svg",
      },
    }));
  });

  it("renders as expected", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
