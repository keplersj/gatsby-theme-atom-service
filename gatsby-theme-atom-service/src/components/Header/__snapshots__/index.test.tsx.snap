// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<div
  className="MuiContainer-root makeStyles-banner-1 MuiContainer-maxWidthXl"
>
  <img
    alt=""
    height="225em"
    src="/static/logo.svg"
    width="225em"
  />
  <h2
    className="MuiTypography-root MuiTypography-h2"
  >
    Example Site
  </h2>
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    This is an example site description.
  </h4>
</div>
`;
